### PR TITLE
Pavel-Latyshou/on Deriv.com for EU and non-EU there is difference in redirecting pop up to the trading platfrom

### PR DESCRIPTION
**Pull request by pavel-latyshou-deriv**
Changes:

-   ...

## Type of change

-   [x ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/localization/localized-link.js [hunk 1]
```diff
@@ -186,12 +185,10 @@ const ExternalLink = ({
     type,
     ...props
 }) => {
-    const { is_eu_country } = useContext(DerivStore)
```
Comment (sean-binary) [inline src/components/localization/localized-link.js:189]: We still need the modal pop for EU when it is an external link (not deriv product links). You need to move smart trader link onto the array that doesn't have the EU modal.

**Review by sean-binary — CHANGES_REQUESTED** [review]

File: src/components/localization/localized-link.js (hunk 1 above)
Comment (sean-binary) [reply src/components/localization/localized-link.js:188]: Should this not be removed anymore then? @pavel-latyshou-deriv

File: src/components/localization/localized-link.js [hunk 2]
```diff
@@ -155,6 +155,8 @@ const getURLFormat = (type, locale, to, affiliate_lang) => {
         return getDerivAppLocalizedURL(localized_link_url[type], locale, to)
     } else if (affiliate_links.includes(type)) {
         return `${localized_link_url[type]}?lang=${affiliate_lang}`
+    } else if (deriv_social_platforms.includes(type) && type === 'smart_trader') {
+        return `${localized_link_url[type]}/${getThaiExcludedLocale(locale)}/${to}.html`
```
Comment (sean-binary) [inline src/components/localization/localized-link.js:159]: The only change that should be required for the PR should be to add `&& !deriv_other_products.includes(type)` on the conditions for `show_modal` on line 192. `smart_trader` shouldn't be categorised under social platforms.

**Review by sean-binary — CHANGES_REQUESTED** [review]

**Review by sean-binary — APPROVED** [review]

**Review by yashim-deriv — CHANGES_REQUESTED** [review]
Is there a reason to remove the `is_eu_country` check completely? To fix the redirect modal, you can simply add `!deriv_other_products.includes(type)` to not show modal for that specific case

**Pull request closed without merging**